src/core/Akka/Actor/Props.cs

```diff
@@ -22,8 +22,76 @@ namespace Akka.Actor
     ///   private Props otherProps = props.WithDeploy(deployment info);
     ///  </code>
     /// </summary>
-    public class Props
+    public class Props : IEquatable<Props>
     {
+        public bool Equals(Props other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return CompareDeploy(other) && CompareSupervisorStrategy(other) && CompareArguments(other) && CompareInputType(other) && CompareProducer(other);
+        }
+
+        private bool CompareInputType(Props other)
+        {
+            return true;
+            return inputType == other.inputType;
+        }
+
+        private bool CompareDeploy(Props other)
+        {
+            return true;
+            return Equals(Deploy, other.Deploy);
+        }
+
+        private bool CompareProducer(Props other)
+        {
+            return true;
+            return Equals(producer, other.producer);
+        }
+
+        private bool CompareSupervisorStrategy(Props other)
+        {
+            return true;
+            return Equals(SupervisorStrategy, other.SupervisorStrategy);
+        }
+
+        private bool CompareArguments(Props other)
+        {
+            return true;
+            if (other == null)
+                return false;
+
+            if (Arguments == null && other.Arguments == null)
+                return true;
+
+            if (Arguments == null)
+                return false;
+
+            return Arguments.SequenceEqual(other.Arguments);
```

Comment:
Err, whoops - that doesn't make sense in any context other than serialization.

The other idea that might be worth considering is doing some reflection magic - traversing down the tree of publicly exposed properties (including nested user-defined objects within nested user-defined objects) and going all the way down until you get to comparable base types.

Or you could have a short-circuit that tests for equality only if the args are a primitive type or explicitly support IEquatable or IComparable. In the event that it's a POCO object without any equality comparer, just make sure that at least the types are identical and that both values are null or both values are not null.